lisa/energy_meter.py

```diff
@@ -407,12 +407,14 @@ def reset(self):
             csv_file = os.path.join(self._res_dir, 'samples_{}.csv'.format(channel))
 
             # Start a dedicated iio-capture instance for this channel
-            self._iio[ch_id] = Popen([self._iiocapturebin, '-n',
+            self._iio[ch_id] = Popen(['stdbuf', '-i0', '-o0', '-e0',
```

Comment:
Unless it's proven that we really need this incantation, I'd happily do without it.

Reply:
@douglas-raillard-arm was willing to keep it I think. I don't mind either way.

Reply:
According the the glibc doc, terminating the process will not cause it to flush streams. Since stdout is a stream and `printf()` is equivalent to `fprintf(stdout)` [1], I can't see anything that would force iio-capture to actually write to stdout before dying (unless it explicitly disable buffering on stdout). Maybe iio-capture is made to be robust, or we were just lucky for now.

[1] http://port70.net/~nsz/c/c11/n1570.html#7.21.6.3p2